MetaMask/metamask-extension
# New reusable gas-display component

## Explanation
Adding new reusable gas componet by extarcting the `renderGasDetailsItem` in ConfirmTransactionBase and replacing the current code from confirm transcation, token approval, and send flow.

Fix: #17459 


## Screenshots/Screencaps



### Before



### After



## Manual Testing Steps
1. Initiate the send flow and, on the send page, verify the correct gas display is showing up for the EIP1559 transaction and legacy transaction.
2. In the confirm page, verify the correct gas display is showing up for the EIP1559 transaction and legacy transaction.
3. In confirm approve and token allowance pages, verify that the correct gas display is showing up for the EIP1559 transactions.


## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
